src/request.rs

```diff
@@ -52,6 +54,52 @@ impl Request {
             sender: Some(sender),
             receiver: Some(receiver),
             local: TypeMap::new(),
+            peer_addr: None,
+            local_addr: None,
+        }
+    }
+
+    #[doc(hidden)]
+    pub fn set_peer_addr(&mut self, peer_addr: Option<impl std::string::ToString>) {
+        self.peer_addr = peer_addr.map(|addr| addr.to_string());
+    }
+
+    #[doc(hidden)]
+    pub fn set_local_addr(&mut self, local_addr: Option<impl std::string::ToString>) {
+        self.local_addr = local_addr.map(|addr| addr.to_string());
+    }
+
+    /// Get the peer socket address for the underlying transport, if appropriate
+    pub fn peer_addr(&self) -> Option<&str> {
+        self.peer_addr.as_deref()
+    }
+
+    /// Get the local socket address for the underlying transport, if appropriate
+    pub fn local_addr(&self) -> Option<&str> {
+        self.local_addr.as_deref()
+    }
+
+    /// Get the remote address for this request.
+    pub fn remote(&self) -> Option<&str> {
+        self.forwarded_for().or(self.peer_addr())
+    }
+
+    /// Parses the Forwarded or X-Forwarded-For headers.
+    /// The returned String will either be an IP address or a domain and an optional port.
+    pub fn forwarded_for(&self) -> Option<&str> {
```

Comment:
This should not be part of the public API. The way to access values produced by this should be through `remote` or `host`.

```suggestion
    fn forwarded_for(&self) -> Option<&str> {
```